crates/noirc_frontend/src/hir/type_check/expr.rs

```diff
@@ -256,22 +256,13 @@ fn type_check_index_expression(
 ) -> Type {
     let index_type = type_check_expression(interner, &index_expr.index, errors);
     let span = interner.expr_span(&index_expr.index);
-
-    index_type.unify(&Type::comp_time(Some(span)), span, errors, || {
-        // Specialize the error in the case the user has a Field, just not a `comptime` one.
-        if matches!(index_type, Type::FieldElement(..)) {
-            TypeCheckError::Unstructured {
-                msg: format!("Array index must be known at compile-time, but here a non-comptime {index_type} was used instead"),
-                span,
-            }
-        } else {
-            TypeCheckError::TypeMismatch {
-                expected_typ: "comptime Field".to_owned(),
-                expr_typ: index_type.to_string(),
-                expr_span: span,
-            }
-        }
-    });
+    if index_type.is_subtype_of(&Type::field(Some(span)), span).is_err() {
```

Comment:
Should use `make_subtype_of` for asserting that something should be a subtype of field.

We could possibly look into removing comptime once this PR is merged, though this is a larger change that we should of course discuss separately.

Reply:
Done